File: tests/typesafety/typescript/exercise_lookup_start.ts
// expect: passes

function findUserId(login: string): string | null {
  return login === "ada" ? "u7" : null;
}

function findCart(userId: string): string[] | null {
  return userId === "u7" ? ["book"] : null;
}

export function firstItem(login: string): string | null {
  const userId = findUserId(login);
  if (userId === null) {
    return null;
  }
  const cart = findCart(userId);
  if (cart === null) {
    return null;
  }
  return cart[0] ?? null;
}

File: tests/typesafety/python/exercise_lookup_start.py
# expect: passes
# title: first_item returns None for three different reasons
def find_user_id(login: str) -> str | None:
    return "u7" if login == "ada" else None


def find_cart(user_id: str) -> list[str] | None:
    return ["book"] if user_id == "u7" else None


def first_item(login: str) -> str | None:
    user_id = find_user_id(login)
    if user_id is None:
        return None
    cart = find_cart(user_id)
    if cart is None:
        return None
    if not cart:
        return None
    return cart[0]
